### ISSUE TITLE
logic_bug lint should understand `cfg!`

### ISSUE THREAD

**Issue by ipetkov** [rust-lang-nursery/rust-clippy#1932]
Had a use case where a certain input parameter must trigger some fallback behavior for a specific OS, but clippy reported it as a logical bug:

```rust
let v: Vec<_> = ...;

if cfg!(windows) && v.is_empty() {
   // fallback behavior
} else {
   // default behavior for Unix/non-empty param
}
```

Clippy is correctly reporting that this is equivalent to `false && ...` on unix systems, but removing the first check (as per the suggestion) does not reflect the actual programmer intent

**Comment by oli-obk** [comment]
Dupe of https://github.com/rust-lang-nursery/rust-clippy/issues/1885

Rustc isn't telling us about macros anymore... we can't do much on our end

**Comment by llogiq** [comment]
Is there a rustic bug about this?

**Comment by oli-obk** [comment]
https://github.com/rust-lang/rust/issues/43268
